src/golang/lib/container_registry/ecr.go

```diff
@@ -0,0 +1,225 @@
+package container_registry
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aqueducthq/aqueduct/lib/errors"
+	"github.com/aqueducthq/aqueduct/lib/lib_utils"
+	"github.com/aqueducthq/aqueduct/lib/models/shared"
+	"github.com/aqueducthq/aqueduct/lib/vault"
+	"github.com/aqueducthq/aqueduct/lib/workflow/operator/connector/auth"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+type ecrCredentials struct {
+	Token         string
+	ExpireAt      int64
+	ProxyEndpoint string
+}
+
+func createAWSSessionFromAccessKey(accessKeyID, secretAccessKey, region string) (*session.Session, error) {
```

Comment:
Can we define these two functions in some shared lib? There are other places where we'll create sessions (I think S3 storage is one of them), and it'll be good to have one set of functions that handle the connection w/ varying credentials.

Reply:
The assumption is a bit different between the two. Specifically, when creating AWS session for S3, we assume we already has `region` available whereas for ECR, we need to retrieve it first. Then the only reusable part seems to be:
```
	sess, err := session.NewSession(&aws.Config{
		Region: aws.String(s3Config.Region),
		Credentials: credentials.NewSharedCredentials(
			s3Config.CredentialsPath,
			s3Config.CredentialsProfile,
		),
	})
```
Which is already a single function call so for now I don't feel super necessary to further put them into a shared lib.